handlers/base.go

```diff
@@ -0,0 +1,18 @@
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+func RespondWithJSON(w http.ResponseWriter, code int, payload interface{}) {
+	response, _ := json.Marshal(payload)
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(code)
+	w.Write(response)
+}
+
+func RespondWithError(w http.ResponseWriter, code int, message string) {
```

Comment:
I think (and that adresses what @abonas asks below) that it could be good to have multiple more specialised functions for cases like not_found, internal_error etc. Those functions should not get a code passed in